chain/network/src/peer_manager.rs

```diff
@@ -228,7 +231,24 @@ impl PeerManagerActor {
         });
     }
 
-    fn remove_active_peer(&mut self, ctx: &mut Context<Self>, peer_id: &PeerId) {
+    /// Remove peer from active set.
+    /// Check it match peer_type to avoid removing a peer that both started connection to each other.
```

Comment:
Does this mean there can be two connections between a pair of peers?

Reply:
There will not be to active connections, but a very reasonable and tricky scenario that needs to be handled is when peer A initiates a connection to B, and at the same time B initiates a connection to A. In that case we accept at most one connection,, but it was handled badly while removing one of the connections. With this change we take care that we remove the connection we meant to remove (outgoing or inbound) and not both.

Reply:
How do we determine which connection we want to remove? Is there a test for this scenario?

Reply:
We choose peer with "highest id" to break ties. https://github.com/nearprotocol/nearcore/blob/master/chain/network/src/peer_manager.rs#L1283
`peer_handshake` test that.